codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/EndpointConfigCustomization.kt

```diff
@@ -38,21 +39,38 @@ internal class EndpointConfigCustomization(
                 "Params" to typesGenerator.paramsStruct(),
             )
             when (section) {
-                is ServiceConfig.ConfigStruct -> rustTemplate(
-                    "pub (crate) endpoint_resolver: $sharedEndpointResolver,",
-                    *codegenScope,
-                )
+                is ServiceConfig.ConfigStruct -> {
+                    if (runtimeMode.defaultToMiddleware) {
+                        rustTemplate(
+                            "pub (crate) endpoint_resolver: $sharedEndpointResolver,",
+                            *codegenScope,
+                        )
+                    }
+                }
 
-                is ServiceConfig.ConfigImpl ->
-                    rustTemplate(
-                        """
-                        /// Returns the endpoint resolver.
-                        pub fn endpoint_resolver(&self) -> $sharedEndpointResolver {
-                            self.endpoint_resolver.clone()
-                        }
-                        """,
-                        *codegenScope,
-                    )
+                is ServiceConfig.ConfigImpl -> {
+                    if (runtimeMode.defaultToOrchestrator) {
+                        rustTemplate(
+                            """
+                            /// Returns the endpoint resolver.
+                            pub fn endpoint_resolver(&self) -> $sharedEndpointResolver {
+                                self.inner.load::<$sharedEndpointResolver>().expect("endpoint resolver should be set").clone()
+                            }
+                            """,
+                            *codegenScope,
+                        )
+                    } else {
+                        rustTemplate(
+                            """
+                            /// Returns the endpoint resolver.
+                            pub fn endpoint_resolver(&self) -> $sharedEndpointResolver {
+                                self.endpoint_resolver.clone()
+                            }
+                            """,
+                            *codegenScope,
+                        )
+                    }
+                }
```

Comment:
Looking at these customizations, they all seem to have the same outer block and different innards. Therefore you could reduce duplication with something like this:

```suggestion
                is ServiceConfig.ConfigImpl -> {
                    docs("Returns the endpoint resolver.")
                    rustBlockTemplate(
                        "pub fn endpoint_resolver(&self) -> $sharedEndpointResolver",
                        *codegenScope,
                    ) {
                        if (runtimeMode.defaultToOrchestrator) {
                            """self.inner.load::<$sharedEndpointResolver>().expect("endpoint resolver should be set").clone()"""
                        } else {
                            "self.endpoint_resolver.clone()"
                        }
                    }
                }
```

It's a bit shorter, although it may make things a little less clear. What do you think?

Reply:
I originally took the exact approach you suggested. Then I realized the clarity suffers a bit, and more importantly, the ease of deleting the middleware code when we switch to the orchestrator. Throughout the PR, we'll see duplicated snippets for the orchestrator and the middleware, but they are "optimized" for future removal.

Reply:
That's smart! I didn't even consider that.